python/ray/ml/predictors/integrations/torch/torch_predictor.py

```diff
@@ -0,0 +1,105 @@
+from typing import Optional, Union, List
+
+import torch
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+
+class TorchPredictor(Predictor):
+    """A predictor for PyTorch models.
+
+    Args:
+        model: The torch module to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+    """
+
+    def __init__(
+        self, model: torch.nn.Module, preprocessor: Optional[Preprocessor] = None
+    ):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model: Optional[torch.nn.Module] = None
+    ) -> "TorchPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model: If the checkpoint contains a model state dict, and not
+                the model itself, then the state dict will be loaded to this
+                ``model``.
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        feature_columns: Optional[
+            Union[List[str], List[List[str]], List[int], List[List[int]]]
+        ] = None,
+        dtype: Optional[torch.dtype] = None,
+    ) -> DataBatchType:
+        """Run inference on data batch.
+
+        The data is converted into a torch Tensor before being inputted to
+        the model.
+
+        Args:
+            data: A batch of input data. Either a pandas DataFrame or numpy
+                array.
+            feature_columns: The names or indices of the columns in the
+                data to use as features to predict on. If this arg is a
+                list of lists, then the data batch will be converted into a
+                multiple tensors which are then concatenated before feeding
+                into the model. This is useful for multi-input models. If
+                None, then use all columns in ``data``.
+            dtype: The torch dtype to use when creating the torch tensor.
```

Comment:
dtype of what? Input is `DataBatchType`, output too. we should say what mean be "the torch tensor". I think there can be multiple tensors, too, as per above "list of lists" definition. should we then also have lists of dtypes?

I fear this might go a little too far, but we need to anticipate questions like this.

Reply:
the multiple tensors are concatenated into a single tensor before being inputted to the model.